src/lib.rs

```diff
@@ -963,6 +989,14 @@ pub fn parse_uri<'a>(bytes: &mut Bytes<'a>) -> Result<&'a str> {
             return Err(Error::Token);
         }
 
+        #[cfg(feature = "utf8_in_path")]
+        // SAFETY: all bytes up till `i` must have been `is_token` and therefore also utf-8.
+        return match str::from_utf8(unsafe { bytes.slice_skip(1) }) {
```

Comment:
This induce an overhead of 30 to 40% perf in uri benches, so not sure if we want this ? but in this case we are not sure that the string is valid utf8 unless using it (see the relative test)

Reply:
```
uri/uri_0001b            1.00      1.1±0.01ns  1786.2 MB/sec    4.58      4.9±0.09ns   390.1 MB/sec
uri/uri_0002b            1.00      1.4±0.01ns     2.1 GB/sec    4.45      6.0±0.34ns   474.7 MB/sec
uri/uri_0004b            1.00      1.9±0.02ns     2.5 GB/sec    4.36      8.1±0.08ns   590.5 MB/sec
uri/uri_0008b            1.00      1.2±0.08ns     6.8 GB/sec    4.98      6.2±0.10ns  1390.9 MB/sec
uri/uri_0016b            1.00      1.8±0.01ns     9.0 GB/sec    2.80      4.9±0.08ns     3.2 GB/sec
uri/uri_0032b            1.00      2.8±0.20ns    11.1 GB/sec    2.05      5.7±0.12ns     5.4 GB/sec
uri/uri_0064b            1.00      5.5±0.75ns    11.1 GB/sec    1.30      7.1±0.04ns     8.5 GB/sec
uri/uri_0256b            1.11     19.1±0.09ns    12.5 GB/sec    1.00     17.2±2.03ns    14.0 GB/sec
uri/uri_0512b            1.36     40.4±0.26ns    11.8 GB/sec    1.00     29.8±3.51ns    16.0 GB/sec
uri/uri_1024b            1.35     80.6±0.53ns    11.8 GB/sec    1.00     59.9±0.34ns    15.9 GB/sec
uri/uri_2048b            1.69    163.7±1.62ns    11.7 GB/sec    1.00     97.1±0.45ns    19.7 GB/sec
uri/uri_4096b            1.95   310.9±22.99ns    12.3 GB/sec    1.00    159.4±0.92ns    23.9 GB/sec
version/http11           1.05      0.8±0.05ns    11.9 GB/sec    1.00      0.7±0.00ns    12.5 GB/sec
```

Reply:
I also try to use [simdutf8](https://github.com/rusticstuff/simdutf8/), but not sure we wants a dep for that, result are better on large string, but no really significant on small string : 

```
group                    no_utf8                                simd_utf8                              utf8
-----                    -------                                ---------                              ----
uri/uri_0001b            1.00      1.1±0.01ns  1786.2 MB/sec    5.42      5.8±0.07ns   329.6 MB/sec    4.58      4.9±0.09ns   390.1 MB/sec
uri/uri_0002b            1.00      1.4±0.01ns     2.1 GB/sec    4.80      6.5±0.02ns   440.1 MB/sec    4.45      6.0±0.34ns   474.7 MB/sec
uri/uri_0004b            1.00      1.9±0.02ns     2.5 GB/sec    4.34      8.0±0.44ns   593.1 MB/sec    4.36      8.1±0.08ns   590.5 MB/sec
uri/uri_0008b            1.00      1.2±0.08ns     6.8 GB/sec    5.58      6.9±0.06ns  1240.3 MB/sec    4.98      6.2±0.10ns  1390.9 MB/sec
uri/uri_0016b            1.00      1.8±0.01ns     9.0 GB/sec    3.47      6.1±0.07ns     2.6 GB/sec    2.80      4.9±0.08ns     3.2 GB/sec
uri/uri_0032b            1.00      2.8±0.20ns    11.1 GB/sec    2.48      6.9±0.12ns     4.5 GB/sec    2.05      5.7±0.12ns     5.4 GB/sec
uri/uri_0064b            1.00      5.5±0.75ns    11.1 GB/sec    1.23      6.7±0.13ns     9.0 GB/sec    1.30      7.1±0.04ns     8.5 GB/sec
uri/uri_0128b            1.21     10.3±0.06ns    11.7 GB/sec    1.00      8.5±1.10ns    14.1 GB/sec    1.19     10.2±0.90ns    11.8 GB/sec
uri/uri_0256b            1.72     19.1±0.09ns    12.5 GB/sec    1.00     11.1±0.30ns    21.5 GB/sec    1.54     17.2±2.03ns    14.0 GB/sec
uri/uri_0512b            2.02     40.4±0.26ns    11.8 GB/sec    1.00     20.0±0.36ns    23.9 GB/sec    1.49     29.8±3.51ns    16.0 GB/sec
uri/uri_1024b            2.48     80.6±0.53ns    11.8 GB/sec    1.00     32.5±1.46ns    29.4 GB/sec    1.84     59.9±0.34ns    15.9 GB/sec
uri/uri_2048b            2.70    163.7±1.62ns    11.7 GB/sec    1.00     60.6±1.06ns    31.5 GB/sec    1.60     97.1±0.45ns    19.7 GB/sec
uri/uri_4096b            2.77   310.9±22.99ns    12.3 GB/sec    1.00    112.4±1.09ns    33.9 GB/sec    1.42    159.4±0.92ns    23.9 GB/sec